uint8 waypoint_index
